### PR TITLE
fix: remove --no-optional from docker-compose build

### PR DESCRIPTION
**Pull request by mistercrunch**
Remove unusual artifact from local docker-compose setup, using vanilla `npm install`

### CODE REVIEW

File: docker/docker-frontend.sh [hunk 1]
```diff
@@ -24,12 +24,15 @@ if [ "$PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" = "false" ]; then
...
+
+    echo "Running `npm install`"
+    npm install
 
     echo "Running frontend"
     npm run dev
+
 else
-    echo "Skipping frontend build steps - YOU RUN IT MANUALLY ON THE HOST!"
+    echo "Skipping frontend build steps - YOU NEED TO RUN IT MANUALLY ON THE HOST!"
```
Comment (rusackas) [inline docker/docker-frontend.sh:37]:
```suggestion
    echo "Skipping frontend build steps - YOU NEED TO RUN IT MANUALLY ON THE HOST! (https://superset.apache.org/docs/contributing/development/#webpack-dev-server)"
```

Comment (rusackas) [reply docker/docker-frontend.sh:37]: Maybe the link will help? ¯\_(ツ)_/¯

File: docker/docker-frontend.sh [hunk 2]
```diff
@@ -24,12 +24,16 @@ if [ "$PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" = "false" ]; then
 fi
 
 if [ "$BUILD_SUPERSET_FRONTEND_IN_DOCKER" = "true" ]; then
+    echo "Building Superset frontend in dev mode inside docker container"
     cd /app/superset-frontend
-    npm install -f --no-optional --global webpack webpack-cli
```
Comment (mistercrunch) [inline docker/docker-frontend.sh:28]: this is not needed

**Review by rusackas — APPROVED** [review]

**Comment by sadpandajoe** [comment]
@supersetbot label 4.1

**Comment by sadpandajoe** [comment]
@supersetbot unlabel 4.1

**Pull request merged**